### PR TITLE
Database and CourseId to TaskDispenser

### PR DESCRIPTION
**Pull request by maleclercq**
Adding a way to access to the database and the courseId from the TaskDispenser

### CODE REVIEW

**Review by Drumor — COMMENTED** [review]
Thank u for this work,really appreciate! Is it useful to change file permissions (644->755) ?

**Comment by maleclercq** [comment]
No the change on file permissions is an error, it shouldn't be on the PR sorry for that

**Comment by maleclercq** [comment]
Now it should be fine. 
I can't sign the CLA with the other account because it is a non-github account

File: inginious/frontend/task_dispensers/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 
 class TaskDispenser(metaclass=ABCMeta):
-    def __init__(self, task_list_func, dispenser_data):
+    def __init__(self, task_list_func, dispenser_data,database=None,courseId=None):
```
Comment (anthonygego) [inline inginious/frontend/task_dispensers/__init__.py:5]: ``course_id`` is more pythonic.

**Comment by maleclercq** [comment]
This should make it more clear

File: inginious/frontend/task_dispensers/__init__.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 
 class TaskDispenser(metaclass=ABCMeta):
-    def __init__(self, task_list_func, dispenser_data):
+    def __init__(self, task_list_func, dispenser_data, database=None, course_id=None):
```
Comment (anthonygego) [inline inginious/frontend/task_dispensers/__init__.py:5]: Making those arguments optional is probably not needed here and has probably no much sense as they will always be passed by design (line 86 in ``inginious/frontend/courses.py``).

The new arguments are also missing from the docstring.

**Comment by anthonygego** [comment]
LGTM. I'll squash this from the GitHub interface but it would be worth learning how to rebase a branch for further contributions ;-)

**Pull request merged**